### PR TITLE
App Bundle reference

### PR DESCRIPTION
**Pull request by ljakimczuk**
### What does this PR do?

This PR explains the App Bundle concept. 

Towards: https://github.com/giantswarm/giantswarm/issues/23587

### Any background context you can provide?

As of now, App Bundles explanations seem to be scattered across different docs and repositories, making them unhandy to work with. This PR is meant to group them in a single place and transform into public doc.

### CODE REVIEW

**Comment by marians** [comment]
A very global one: we Want to use sentence case throughout our docs. I see no reason to capitalize "App Bundle". "App" should only be titlecased if it's about the CR, when referencing the kind name.

File: src/content/app-platform/app-bundle/index.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+  values: |
+    clusterName: wc1
+    organization: testorg
+kind: ConfigMap
+metadata:
+  name: bundle-wc1-userconfig
+  namespace: wc1
+```
+
+Please refer to the [App Configuration]({{< relref "/app-platform/app-configuration" >}}) to understand how the
```
Comment (marians) [inline src/content/app-platform/app-bundle/index.md:237]:
```suggestion
Please refer to [App Configuration]({{< relref "/app-platform/app-configuration" >}}) to understand how the
```

File: src/content/app-platform/app-bundle/index.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+user configuration works, and to the [kubectl gs template app]({{< relref "/ui-api/kubectl-gs/template-app.md" >}}) to get of how you can automate this process.
```
Comment (marians) [inline src/content/app-platform/app-bundle/index.md:238]:
```suggestion
user configuration works, and to the [kubectl gs template app]({{< relref "/ui-api/kubectl-gs/template-app.md" >}}) reference to understand how you can automate this process.
```

File: src/content/app-platform/app-bundle/index.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+linkTitle: App Bundle
+title: App Bundle reference
+description: Overview of the the App Bundle concept, how it works, and how to configure it.
+menu:
+  main:
+    parent: app-platform
+weight: 30
+aliases:
+ - /reference/app-bundle/
+last_review_date: 2022-09-14
```
Comment (marians) [inline src/content/app-platform/app-bundle/index.md:11]: Aliases should only be created if a page is moved, to have a redirect from the old URL (the alias) to the new URL.

```suggestion
weight: 30
last_review_date: 2022-09-14
```

File: src/content/app-platform/app-bundle/index.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-honeybadger
+user_questions:
+ - What is the App Bundle?
+ - How does the App Bundle installation work?
```
Comment (marians) [inline src/content/app-platform/app-bundle/index.md:16]:
```suggestion
 - How does installation of an app bundle work?
```

File: src/content/app-platform/app-bundle/index.md [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+ - How to properly configure the App Bundle?
```
Comment (marians) [inline src/content/app-platform/app-bundle/index.md:17]:
```suggestion
 - How to properly configure an app bundle?
```

File: src/content/app-platform/app-bundle/index.md [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+ - Are there any caveats in using App Bundle?
```
Comment (marians) [inline src/content/app-platform/app-bundle/index.md:18]:
```suggestion
 - Are there any caveats in using app bundles?
```

File: src/content/app-platform/app-bundle/index.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+  name: bundle-app
+  namespace: wc1
+  ...
+```
+
+Secondly, each app bundle Helm Chart exposes the `clusterName` and `organization` configuration options in its
+`values.yaml` file, see below.
+
+```yaml
+clusterName: ""
```
Comment (QuentinBisson) [inline src/content/app-platform/app-bundle/index.md:196]: My only question is: Why did we move to clusterName instead of clusterID ?

**Comment by QuentinBisson** [comment]
It looks great :)

File: src/content/app-platform/app-bundle/index.md (hunk 7 above)
Comment (ljakimczuk) [reply src/content/app-platform/app-bundle/index.md:196]: I don't remember of when exactly this decision has been taken, but I doubt it was one team behind it as there seem to be a global movement towards cluster names. At least I've seen it being used all over the places.

**Review by QuentinBisson — APPROVED** [review]

**Review by mproffitt — APPROVED** [review]

**Pull request merged**